platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/StateLifecycleManager.java

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.swirlds.state;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+
+/**
+ * Implementations of this interface are responsible for managing the state lifecycle:
+ * <ul>
+ * <li>Maintaining references to a mutable state and the latest immutable state.</li>
+ * <li>Creating snapshots of the state.</li>
+ * <li>Loading snapshots of the state.</li>
+ * <li>Creating a mutable copy of the state, while making the current mutable state immutable.</li>
+ * </ul>
+ * @param <T> A type of the snapshot source, which should implement {@link MerkleNodeStateAware}.
+ */
```

Comment:
What about threading? What methods of this interface can be called on what threads?